ConfigSpace/util.pyx

```diff
@@ -41,6 +38,9 @@ from ConfigSpace.hyperparameters import CategoricalHyperparameter, \
 import ConfigSpace.c_util
 cimport cython
 
+# OPTIM: To prevent large memory allocations, we place an upperbound on the maximum
+# amount of neighbors to be sampled
+MAX_NEIGHBORHOOD = 10_000

```

Comment:
There was an issue here where the number of requested neighbors for a `UniformIntegerHyperparameter` was set to the full possible range. This would firstly blow up memory as it would try to generate every possible neighbor and secondly, only a fraction of these were used.

This constant can be seen in use later on in this PR.